packages/jest-core/src/SearchSource.ts

```diff
@@ -37,8 +37,55 @@ export type TestSelectionConfig = {
   watch?: boolean;
 };
 
-const globsToMatcher = (globs: Array<Config.Glob>) => (path: Config.Path) =>
-  micromatch([replacePathSepForGlob(path)], globs, {dot: true}).length > 0;
+const globsMatchers = new Map<
```

Comment:
how does this work with watch mode, and changing the globs? I guess since the key is the glob itself and we don't collect all entries in the `Map` we won't pull stale data out?

Reply:
Yep, that's my understanding!